### PR TITLE
Booknotes modal accessibility updates

### PR DESCRIPTION
**Pull request by bpmcneilly**
This PR introduces some minor accessibility tweaks to the code implemented with PR #4782.

### Technical
The `<details>` & `<summary>` elements were already accessible to ATs, so the primary change was adding a `<fieldset>` element to group all of the options for a given question & then place the question itself in a `<legend>` element as the first child of the `<fieldset>`. This will ensure that the text in the `<legend>` is rendered to ATs when users navigate the groups of form fields.

While I agree that there should be `<h3>` elements, I moved this to be nested as a child of the `<details>` element. Placing the element here, rather than as a child of the `<legend>` will mean that the heading layout will be something like:

- Observations, Heading
  - Pace, Heading
    - What is the pace of this book?
    - [Pace Options]
  - Genres, Heading
    - What are the genres of this book?

rather than:

- Observations, Heading
  - Pace
  - What is the pace of this book?, Heading
    - [Pace Options]
    - Genres
  - What are the genres of this book?, Heading

Some CSS has been added to make no visually breaking changes after the markup has been implemented.

The only outstanding issue from this PR is the color contrast of the expanded or hovered `<summary>` content. Using `@grey` (`#666666`) text on a `@blue-204764` (`#c7d1d8`) background creates a color contrast ratio of 3.7:1 (screenshot included below). According to WCAG, this would be compliant for large sized text, but not regular text. This leaves us with two options, either enlarging the `<summary>` content to become large, or adjusting the colors used.

Currently, the text is just under 14pt in font-size & as it is already bolded, bumping this up to 14pt would qualify it as large text & make the contrast ratio sufficient under WCAG. However, a more robust approach would likely be to darken the grey text. Even the next darkest grey option `@dark-grey-two` would create a color contrast ratio over 4.5:1. We can discuss the best case here, just wanted to flag that issue as outstanding.

### Testing
Verify that these code changes cause no breakage - particularly visually, but also potentially in terms of form handling.

### Screenshot
The booknotes modal dialog with MS Accessibility Insights displaying a contrast ratio of 3.702:1 for expanded summary content

### Stakeholders
@mekarpeles

### CODE REVIEW

File: openlibrary/plugins/openlibrary/js/patron-metadata/index.js [hunk 1]
```diff
@@ -30,11 +30,11 @@ export function initPatronMetadata() {
 
             $form.append(`
               <details class="aspect-section">
-                <summary>${aspect.label}</summary>
-                <div id="${aspect.label}-question">
-                    <h3>${aspect.description}</h3>
+                <summary><h3>${aspect.label}</h3></summary>
+                <fieldset id="${aspect.label}-question">
+                    <legend>${aspect.description}</legend>
                     ${$choices.prop('outerHTML')}
```
Comment (jdlrobson) [inline openlibrary/plugins/openlibrary/js/patron-metadata/index.js:36]: Note to self: not related to this change, but not sure why this line is using prop outerHTML and not outerHTML directly...

**Review by jdlrobson — COMMENTED** [review]
Code looks fine. How do I test the patron meta data locally? What do I need to do and which URL do I go to?

**Comment by jamesachamp** [comment]
@jdlrobson In order to test, you'll first have to add your account to the "beta-testers" usergroup.  To add group members locally, navigate here:
http://localhost:8080/usergroup/beta-testers?m=edit
Then, add the accounts in the "Members" fields and click the save button.  Members must begin with "/people/". For example, "/people/openlibrary" adds user "openlibrary" to the group.

Once this is done, you can access the modal from any book page.  Clicking on the "My book notes" link will open the modal:

my_book_notes_link

**Review by jdlrobson — COMMENTED** [review]
looks good. I just need to take the time to get a working environment to test this.

**Comment by jdlrobson** [comment]
Sorry I'm set up for testing now but this has a merge conflict that needs resolving.

**Review by jdlrobson — APPROVED** [review]
after fixing rebase conflict this is good to go

**Pull request closed without merging**